games/openttd/patch-build.diff

```diff
@@ -0,0 +1,36 @@
+diff --git config.lib config.lib
+index 304b86e..03c1536 100644
+--- config.lib
++++ config.lib
+@@ -667,7 +667,7 @@ check_params() {
+ 
+ 	if [ "$enable_lto" != "0" ]; then
+ 		# GCC 4.5 outputs '%{flto}', GCC 4.6 outputs '%{flto*}'
+-		has_lto=`($cxx_build -dumpspecs 2>&1 | grep '\%{flto') || ($cxx_build -help ipo 2>&1 | grep '\-ipo')`
++		has_lto=`($cxx_build -dumpspecs 2>&1 | grep '\%{flto') || ($cxx_build -help ipo 2>&1 | grep '\-ipo') || ($cxx_build --help 2>&1 | grep 'lto.*thin')`
+ 		if [ -n "$has_lto" ]; then
+ 			log 1 "using link time optimization... yes"
+ 		else
+@@ -1336,6 +1336,12 @@ make_compiler_cflags() {
+ 		# rdynamic is used to get useful stack traces from crash reports.
+ 		ldflags="$ldflags -rdynamic"
+ 
++		if [ "$enable_lto" != "0" ]; then
++			flags="$flags -flto=thin"
++			features="$features lto"
++			features_host="$features_host lto"
++		fi
++
+ 	# Assume gcc, since it just uses argv[0] in its --version output
+ 	else
+ 		# Enable some things only for certain GCC versions
+@@ -2104,8 +2110,7 @@ check_osx_sdk() {
+ 			return 1
+ 		fi
+ 
+-		# Set minimum version to 10.4 as that's when kCGBitmapByteOrder32Host was introduced
+-		sysroot="-isysroot $osx_sdk_path -Wl,-syslibroot,$osx_sdk_path -mmacosx-version-min=10.4"
++		sysroot="-isysroot $osx_sdk_path -Wl,-syslibroot,$osx_sdk_path"
```

Comment:
Can you add a comment explaining why building this with `-mmacosx-version-min=10.4` no longer works?

Reply:
Well, it's not that it doesn't work; I just figured that targeting the current OS — as usual — was advantageous?